extension/src/popup/components/manageNetwork/NetworkForm/index.tsx

```diff
@@ -23,60 +38,202 @@ interface FormValues {
   networkName: string;
   networkPassphrase: string;
   networkUrl: string;
-  isSwitchSelected: boolean;
+  isSwitchSelected?: boolean;
 }
 
 interface NetworkFormProps {
   isEditing: boolean;
 }
 
+export const NETWORK_INDEX_SEARCH_PARAM = "networkIndex";
+
 export const NetworkForm = ({ isEditing }: NetworkFormProps) => {
   const { t } = useTranslation();
   const dispatch: AppDispatch = useDispatch();
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const networksList = useSelector(settingsNetworksListSelector);
+  const settingsError = useSelector(settingsErrorSelector);
+  const [isNetworkInUse, setIsNetworkInUse] = useState(false);
+  const [isConfirmingRemoval, setIsConfirmingRemoval] = useState(false);
+  const history = useHistory();
+  const { search } = useLocation();
+
+  const networkIndex = new URLSearchParams(search).get("networkIndex");
+  const networkDetailsToEdit = networksList[Number(networkIndex)];
```

Comment:
If we're editing a Network's config, we grab the Network's index in the NetworksList from a query param